datafeeder/src/main/java/org/georchestra/datafeeder/batch/service/DataUploadAnalysisService.java

```diff
@@ -175,6 +166,43 @@ public DatasetUploadState analyze(DatasetUploadState item) throws Exception {
         return item;
     }
 
+    private Map<String, String> options(DatasetUploadState item) throws Exception {
+        if (item.getFormat() == DataSourceMetadata.DataSourceType.CSV) {
+            return analyzeCsv(item.getAbsolutePath());
+        }
+        return Map.of();
+    }
+
+    @VisibleForTesting
+    public Map<String, String> analyzeCsv(String path) throws Exception {
+        Map<String, String> options = new HashMap<>();
+        options.put("quoteChar", "\"");
+        options.put("delimiter", ",");
+        options.put("csv", csvSixLinesAsBase64(path));
+        return options;
+
+    }
+
+    @VisibleForTesting
+    public String csvSixLinesAsBase64(String path) throws Exception {
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+        try (FileInputStream fis = new FileInputStream(path)) {
+            int numLines = 0;
+            while (true) {
+                int b = fis.read();
+                if (b == '\n' || b == '\r') {
+                    numLines++;
```

Comment:
thinking of this, I'm wondering if the CSV being provided is a Windows-like text file ('\r\n'), then maybe we will always return 3 lines instead of 6.

Reply:
I've pushed a sort of solution to it. With a continue on `\r` char, it should handle CRLF and LF correctly.

Reply:
ok, sounds legit